docs/tutorials/customSessionKeys/createSession.md

```diff
@@ -0,0 +1,253 @@
+
+
+```typescript
+const sessionModule = await SessionKeyManagerModule.create({
+	moduleAddress: DEFAULT_SESSION_KEY_MANAGER_MODULE,
+	smartAccountAddress: address,
+	sessionStorageClient: sessionFileStorage
+});
+
+```
+
+now, we will cover a deployed contract that validates specific permissions to execute ERC20 token transfers. Using this ERC20 Validation module you will be able to create a dApp that allows user to send a limited amount of funds to a specific address without needing to sign a transaction every single time. In the following example user can only transfer 50 tokens at once.
+
+```typescript
+const sessionKeyData = defaultAbiCoder.encode(
+	["address", "address", "address", "uint256"],
+	[sessionKeyEOA,
+	"0xdA5289fCAAF71d52a80A254da614a192b693e977", // erc20 token address
+	"0x322Af0da66D00be980C7aa006377FCaaEee3BDFD", // receiver address
+	ethers.utils.parseUnits("50".toString(), 6).toHexString(), // 50 usdc amount
+	]
+);
+
+```
+
+next we create the session data. We specify how long this should be valid until or how long it is valid after. This should be a unix timestamp to represent the time. Passing 0 on both makes this session never expire, do not do this in production. Next we pass the module address, session key address, and pass the session key data we just created.
+
+
+```typescript
+const sessionTxData = await sessionModule.createSessionData([{
+	validUntil: 0,
+	validAfter: 0,
+	sessionValidationModule: erc20ModuleAddr,
+	sessionPublicKey: sessionKeyEOA,
+	sessionKeyData: sessionKeyData,
+}]);
+```
+We're going to be tracking if the session key module is already enabled. If we need to enable session key module, we create a transaction using the getEnableModuleData and pass the session key manager module address and push this to the array.
+
+
+```typescript
+const isEnabled = await smartAccount.isModuleEnabled(DEFAULT_SESSION_KEY_MANAGER_MODULE)
+if (!isEnabled) {
+	const enableModuleTrx = await smartAccount.getEnableModuleData(DEFAULT_SESSION_KEY_MANAGER_MODULE);
+	transactionArray.push(enableModuleTrx);
+}
+		
+```
+
+Next we will build a userOp and use the smart account to send it to Bundler.
+
+```typescript
+let partialUserOp = await smartAccount.buildUserOp(transactionArray, {
+	paymasterServiceData: {
+		mode: PaymasterMode.SPONSORED,
+	}
+});
+
+const userOpResponse = await smartAccount.sendUserOp(partialUserOp);
+console.log(`userOp Hash: ${ userOpResponse.userOpHash }`);
+
+const transactionDetails = await userOpResponse.wait();
+console.log("txHash", transactionDetails.receipt.transactionHash);
+```
+Now with this implemented, let's take a look at executing the ERC20 token transfer with this session in the next section.
```

Comment:
You can mention that once this transaction is successfully mined it means it enabled session on chain. 
now anyone holding session private key can make use of this session (within provided time range and permissions context)